proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,452 @@
+   maintainers. The current status should be clearly indicated and the next
+   steps well defined.
+
+## Specification
+
+ ![Figure A: Flowchart of contribution guide review
+process.](contrib-guide-high-level.png "Figure A: Flowchart of contribution
+guide review process.")
+
+**Figure A: Flowchart of contribution guide review process.**
+
+To improve the review process for feature requests and pull requests, we propose
+to modify the existing stages and expand them. Figure A demonstrates how a
+request and contribution will proceed through this improved process. The process
+extends and clarifies the three existing stages: 
+
+1. Intake, where feature requests are triaged and those suitable for community
+   contribution are identified;
+2. Implementation, where a contributor acknowledges that they will work on an
+   issue and open a pull request, and
+3. Review, where pull requests are prioritized for review and the maintainers
+   work with the contributor to resolve the contribution.
+
+In addition, we propose to publish an improved contributing guide to document
+the review process, acceptance criteria, and expectations for communication. In
+addition, a transparent status must be published to indicate what is available
+for contribution and how close a contribution is to completion.
+### Intake
+
+The intake process identifies feature requests, through GitHub issues, where
+community contributions are welcome. The maintainers will actively review GitHub
+issues for feature requests to make this determination. Depending on the issue
+type (bug or feature request) and area (code or documentation), a set of
+criteria must be met before marking it as ready for contribution. The criteria
+used to make that decision may include:
+
+1. The change is feasible for a community member to contribute. See the appendix
+   for a definition of what would not be feasible for community contribution.
+1. There is enough detail provided for the maintainers to agree with the
+   implementation plan.
+
+Specifically for feature requests, the change must impact a minimum of 10 of
+users, measured by GitHub reactions.
+
+Once these criteria have been met, an issue is available for the community to
+contribute. The issue will be labeled and added to a queue where users
+interested in contributing can discover what is available.
+
+All issues are subject to existing automated issue closure mechanisms, which
+will prompt for community input before they are automatically closed if there
+are not enough upvotes ([TBD]) and no activity (comments or upvotes) for [TBD;
+currently a year].
+
+### Implementation
+
+Any issue that has been marked as available is open for a community contribution
+through a GitHub pull request. A user can comment that they would like to
+contribute to a specific issue. They then start implementing using the
+CONTRIBUTING guide as a reference. They open a pull request linked to the issue.
```

Comment:
It would be great if we can be a bit more explicit in indicating the mechanism on which an issue is moved out of the "contribution ready" queue (based on the flow chart). For example, something like:

An issue moves from the contribution-ready phase to the implementation phase when a contributor opens a pull request for the issue.

And just keep it at that. Right now it is not too clear what the minimum is required to move to this phase.

Reply:
Yep. I think there's some value in calling out encouraging collaboration and communication on the issue, but not requiring.